projects/src/main/java/org/odk/collect/projects/AddProjectDialog.kt

```diff
@@ -39,17 +40,28 @@ class AddProjectDialog : MaterialFullScreenDialogFragment() {
         super.onViewCreated(view, savedInstanceState)
         setUpToolbar()
 
-        binding.projectIconInputText.addTextChangedListener(OneSignTextWatcher(binding.projectIconInputText))
+        binding.urlInputText.doOnTextChanged { text, _, _, _ ->
+            if (Validator.isUrlValid(text.toString())) {
+                binding.urlInputText.error = null
+                binding.addButton.isEnabled = true
+                binding.addButton.alpha = 1f
+            } else {
+                binding.urlInputText.error = getString(R.string.url_error)
+                binding.addButton.isEnabled = false
+                binding.addButton.alpha = 0.3f
+            }
+        }
 
         binding.cancelButton.setOnClickListener {
             dismiss()
         }
 
         binding.addButton.setOnClickListener {
-            val newProject = Project.New(getProjectName(), getProjectIcon(), getProjectColor())
+            val generatedProjectDetails = ProjectDetailsGenerator.generateProjectDetails(getUrl())
+            val newProject = Project.New(generatedProjectDetails.projectName, generatedProjectDetails.projectIcon, generatedProjectDetails.projectColor)
             val savedProject = projectsRepository.save(newProject)
 
-            listener?.onProjectAdded(savedProject)
+            listener?.onProjectAdded(savedProject, getUrl(), getUsername(), getPassword())
```

Comment:
You know what... I was thinking about it but I'm not sure I would leave it as it is for now since:
1. it's not something very important
2. it is covered with tests (so even if it's not a perfect solution at least it's testable)

let's see what other prs require maybe it will make it clearer.

Reply:
That sounds like it makes sense